File: CS/T191577/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using T191577.Reports;
using DevExpress.Web.Mvc;

namespace T191577.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult DocumentViewerPartial() {
            CategoriesReport report = new CategoriesReport();
            return PartialView("_DocumentViewerPartial", report);
        }

        public ActionResult DocumentViewerExport() {
            CategoriesReport report = new CategoriesReport();
            return DocumentViewerExtension.ExportTo(report);
        }
    }
}

File: VB/T191577/Controllers/HomeController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports T191577.Reports
Imports DevExpress.Web.Mvc

Namespace T191577.Controllers
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function DocumentViewerPartial() As ActionResult
            Dim report As New CategoriesReport()
			Return PartialView("_DocumentViewerPartial", report)
		End Function

		Public Function DocumentViewerExport() As ActionResult
			Dim report As New CategoriesReport()
			Return DocumentViewerExtension.ExportTo(report)
		End Function
	End Class
End Namespace